src/common/runtime/src/repeated_task.rs

```diff
@@ -42,14 +43,69 @@ struct TaskInner<E> {
     task_handle: Option<JoinHandle<()>>,
     /// The task_fn to run. This is Some if the task is not started.
     task_fn: Option<BoxedTaskFunction<E>>,
+    /// Generates the next interval.
+    interval_generator: Option<Box<dyn IntervalGenerator>>,
+}
+
+pub trait IntervalGenerator: Send + Sync {
+    fn next(&mut self) -> Duration;
+    fn is_regular(&self) -> bool {
+        true
+    }
```

Comment:
```suggestion
    fn next(&mut self) -> Duration;
    
    fn is_regular(&self) -> bool {
        true
    }
```

Reply:
BTW, How about allowing the `IntervalGenerator` to return self's display info?
Such as: 

```
pub trait IntervalGenerator: Send + Sync {
    fn next(&mut self) -> Duration;

    fn display(&self) -> String;
```